pkg/apis/datavolumecontroller/v1alpha1/types.go

```diff
@@ -42,9 +42,10 @@ type DataVolumeSpec struct {
 
 // DataVolumeSource represents the source for our Data Volume, this can be HTTP, S3 or an existing PVC
 type DataVolumeSource struct {
-	HTTP *DataVolumeSourceHTTP `json:"http,omitempty"`
-	S3   *DataVolumeSourceS3   `json:"s3,omitempty"`
-	PVC  *DataVolumeSourcePVC  `json:"pvc,omitempty"`
+	HTTP  *DataVolumeSourceHTTP `json:"http,omitempty"`
+	S3    *DataVolumeSourceS3   `json:"s3,omitempty"`
+	PVC   *DataVolumeSourcePVC  `json:"pvc,omitempty"`
+	LOCAL string                `json:"local,omitempty"`
```

Comment:
I would change LOCAL to ```UPLOAD *bool `json:"upload,omitempty"```

Also, since I'm here and you were wondering why code zz_generated.deepcopy.go wasn't updated for `LOCAL` it's because it wasn't necessary.  It's not a pointer.  The following line would copy it:

https://github.com/kubevirt/containerized-data-importer/blob/master/pkg/apis/datavolumecontroller/v1alpha1/zz_generated.deepcopy.go#L91

Change type to `*bool` and code generation will get updated

Reply:
or maybe even just bool

Reply:
You should use a pointer to be consistent and signify that the value is optional.  To me bool makes more sense than string.